src/ILLink.RoslynAnalyzer/RequiresAssemblyFilesAnalyzer.cs

```diff
@@ -46,9 +72,36 @@ public override void Initialize (AnalysisContext context)
 				if (string.Equals (includesAllContent?.Trim (), "true", StringComparison.OrdinalIgnoreCase))
 					return;
 
+				var propertiesBuilder = ImmutableArray.CreateBuilder<IPropertySymbol> ();
+				var methodsBuilder = ImmutableArray.CreateBuilder<IMethodSymbol> ();
+
+				var assemblyType = compilation.GetTypeByMetadataName ("System.Reflection.Assembly");
+				if (assemblyType != null) {
+					// properties
+					AddIfNotNull (propertiesBuilder, TryGetSingleSymbol<IPropertySymbol> (assemblyType.GetMembers ("Location")));
+
+					// methods
+					methodsBuilder.AddRange (assemblyType.GetMembers ("GetFile").OfType<IMethodSymbol> ());
+					methodsBuilder.AddRange (assemblyType.GetMembers ("GetFiles").OfType<IMethodSymbol> ());
+				}
+
+				var assemblyNameType = compilation.GetTypeByMetadataName ("System.Reflection.AssemblyName");
+				if (assemblyNameType != null) {
+					AddIfNotNull (propertiesBuilder, TryGetSingleSymbol<IPropertySymbol> (assemblyNameType.GetMembers ("CodeBase")));
+					AddIfNotNull (propertiesBuilder, TryGetSingleSymbol<IPropertySymbol> (assemblyNameType.GetMembers ("EscapedCodeBase")));
+				}
+
+				var properties = propertiesBuilder.ToImmutable ();
+				var methods = methodsBuilder.ToImmutable ();
+
 				context.RegisterOperationAction (operationContext => {
 					var methodInvocation = (IInvocationOperation) operationContext.Operation;
-					CheckCalledMember (operationContext, methodInvocation.TargetMethod);
+					var targetMethod = methodInvocation.TargetMethod;
+					if (!operationContext.ContainingSymbol.HasAttribute (RequiresAssemblyFilesAttribute) && Contains (methods, targetMethod, SymbolEqualityComparer.Default)) {
```

Comment:
Nit: maybe add a comment that this check and the one we have for properties can be removed once/if we annotate dotnet/runtime?

Reply:
Even with the annotation in dotnet/runtime this code still needs to exist, the description provided by IL3000 and IL3001 is something the users can action and we already have documented. We don't want to produce IL3002 for everything, so these warnings (IL3000 and IL3001) still need to exist and be triggered in case someone calls directly the public API. The only difference would be that they can be suppressed by using the RequiresAssemblyFiles attribute.

Reply:
We can always decide on which diagnostic to print inside CheckCalledMember, also `RequiresAssemblyFilesAttribute` has a `Message` property that can give further instructions to users.